apps/namegraph.dev/lib/utils.ts

```diff
@@ -1,6 +1,126 @@
 import { clsx, type ClassValue } from "clsx";
 import { twMerge } from "tailwind-merge";
+import {
+  NameGraphGroupedByCategoryResponse,
+  NameGraphGroupingCategory,
+  NameGraphSuggestion,
+} from "@namehash/namegraph-sdk/utils";
+import { createNameGraphClient } from "@namehash/namegraph-sdk";
 
 export function cn(...inputs: ClassValue[]) {
   return twMerge(clsx(inputs));
 }
+
+interface LabelAndModeQueryParams {
+  label: string;
+  mode: "full";
+}
+
+export interface NameGraphRelatedCollection {
+  collection_id: string;
+  collection_title: string;
+  collection_members_count: number;
+}
+
+export interface NameGraphSuggestionCategory {
+  type: NameGraphGroupingCategory;
+  // NameGraph API guarantees "name" will always be a normalized name
+  name: string;
+  // NameGraph API guarantees "suggestions" will never be an empty list
+  suggestions: NameGraphSuggestion[];
+  related_collections: NameGraphRelatedCollection[];
+}
+
+// This array is used for checking when a category is not a "related" category
+const NameGraphSuggestionCategoryTypes = Object.values(
+  NameGraphGroupingCategory,
+);
+
+/**
+ * Common usage NameGraphClient
+ */
+const NameGraphClient = createNameGraphClient();
+
+/**
+ * Main function for fetching name suggestions from the NameGraph API.
+ */
+export const generateNamesByQuery = async (
+  input: LabelAndModeQueryParams,
+): Promise<NameGraphSuggestion[]> => {
+  if (input.label.includes("."))
+    throw new Error("Invalid label for generating name suggestions");
+
+  // Related collections params
+  const enable_learning_to_rank = true;
+  const max_names_per_related_collection = 10;
+  // Determines the number of collections returned in category.related_collections
+  const max_recursive_related_collections = 3;
```

Comment:
Please note: I shared another comment in another file how you need to use named constants and not magic numbers. You defined a magic number 3 somewhere and that named constant should be the same as this named constant. They are the same idea!